### ISSUE TITLE
Indexing via CLI leading Slash in URL missing

### ISSUE THREAD

**Issue by honsa** [hmmh/typo3-solr-file-indexer#57]
We use a fresh installation of  TYPO3 12
"typo3/cms-core": "12.4.22"
"apache-solr-for-typo3/solr": "12.0.5"
"hmmh/solr-file-indexer": "3.0.0"
"apache-solr-for-typo3/tika": "12.0.2"

If we execute the indexer via CLI we are missing the leading slash in the URL.
`vendor/bin/typo3 scheduler:run --task=1 -f`

In web context the leading slash is added to the URL.

![image](https://github.com/user-attachments/assets/b5b82d93-bcf1-4788-9605-614f93043442)

Are we missing a setting?

I have recognized in the Web Context the URL is generated using the `TYPO3_SITE_PATH`  constant in the Solr extension. May this has not directly to do something with your Extension but happens only with Files.

Here are two examples how the file is in the Solr index with different URLs depending on how the Task is executed.

Indexing via  CLI:
```
"response":{
    "numFound":1,
    "start":0,
    "numFoundExact":true,
    "docs":[{
      "id":"4e969430583297ad437c4325b4107f6a6fcb1b1d/sys_file_metadata/2",
      "type":"sys_file_metadata",
      "site":"typo3-12.test",
      "siteHash":"4e969430583297ad437c4325b4107f6a6fcb1b1d",
      "uid":2,
      "pid":0,
      "variantId":"712118c994b93fea449d531ebd2dc3f3a4b96625/sys_file_metadata/2",
      "created":"2024-06-04T11:26:18Z",
      "changed":"2024-06-04T11:26:18Z",
      "access":["r:0"],
      "size_intS":34262,
      "fileExtension":"pdf",
      "title_stringS":"Test2.pdf",
      "content":"Test",
      "contentExact":"Test",
      "teaser":"Test",
      "url":"fileadmin/user_upload/Test2.pdf",
      "_version_":1815058020572332032,
      "appKey":"EXT:solr"
    }]
  }
} 
```
Indexing manual via Backend interface:
```
"response":{
    "numFound":1,
    "start":0,
    "numFoundExact":true,
    "docs":[{
      "id":"4e969430583297ad437c4325b4107f6a6fcb1b1d/sys_file_metadata/2",
      "type":"sys_file_metadata",
      "site":"typo3-12.test",
      "siteHash":"4e969430583297ad437c4325b4107f6a6fcb1b1d",
      "uid":2,
      "pid":0,
      "variantId":"712118c994b93fea449d531ebd2dc3f3a4b96625/sys_file_metadata/2",
      "created":"2024-06-04T11:26:18Z",
      "changed":"2024-06-04T11:26:18Z",
      "access":["r:0"],
      "size_intS":34262,
      "fileExtension":"pdf",
      "title_stringS":"Test2.pdf",
      "content":"Test",
      "contentExact":"Test",
      "teaser":"Test",
      "url":"/fileadmin/user_upload/Test2.pdf",
      "_version_":1815058306516910080,
      "appKey":"EXT:solr"
    }]
  }
}
```

**Comment by swilking** [comment]
Hm, we will check this. The URL is generated by the core function TYPO3\CMS\Core\Resource\ResourceStorage::getPublicUrl(), not by our extension.

**Comment by swilking** [comment]
Fixed in 3.0.1 and 4.0.0

**Comment by honsa** [comment]
Thank you for the fix.

May you can use the [TYPO3_SITE_PATH](https://github.com/search?q=repo%3ATYPO3%2Ftypo3%20TYPO3_SITE_PATH%20&type=code) instead of a separate config option.
